src/test/clojure/clara/test_rules.clj

```diff
@@ -2748,3 +2752,26 @@
                     (fire-rules))]
 
     (is (= [{:?x inner-one}] (query session test-query)))))
+
+(deftest test-maximum-forms-per-eval
```

Comment:
@WilliamParker, A test demonstrating the batch eval failure with a method size too large.

This test is a little slow, i was considering putting it in the performance tests, but i would be worried that those test would not be run and we might miss changes to compilation that might break the limit.

Reply:
How long does this take to run?  Perhaps it should be in another test file - it would still run during Leiningen builds but not as part of the test-rules namespace.  I've been trying to move away from having everything in test-rules from a code organization POV anyway.

Reply:
Yikes, that is quite a long-running test.  Perhaps we could put it in a "long_running_tests" file that wouldn't be part of a normal REPL workflow; if there are other instances of things like this it might be worthwhile to create a test profile that is run in Travis but exclude the file from "lein test" runs.

It also seems like this could assert on the exception message added about message size too large?  Or by the exception info, you could add a key if needed.  It would be pretty simple to just recursively get all the messages and make sure one of them matches a regex.